example-app-similar-products-embeddings/src/pages/api/seed.ts

```diff
@@ -0,0 +1,283 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { createClient } from '@/lib/create-graphq-client';
+import { getDatabase } from '@/lib/file-database';
+import { EmbeddingService } from '@/lib/embeddings';
+import { saleorApp } from '@/saleor-app';
+import { gql } from 'urql';
+
+// Configuration for parallel processing
+const CONCURRENCY_LIMIT = parseInt(process.env.EMBEDDING_CONCURRENCY || '3', 10); // Lower for API to be more conservative
+const BATCH_SIZE = parseInt(process.env.EMBEDDING_BATCH_SIZE || '20', 10);
+
+/**
+ * Process a single product and generate its embedding
+ */
+async function processProduct(
+  product: Product,
+  embeddingService: EmbeddingService,
+  db: any
+): Promise<{ success: boolean; skipped: boolean; error?: string }> {
+  try {
+    console.log(`Processing: ${product.name} (${product.id})`);
+
+    const embedding = await embeddingService.generateEmbedding(product);
+    if (!embedding) {
+      console.log(`Skipped: No usable content for embedding`);
+      return { success: false, skipped: true };
+    }
+
+    await db.upsertProduct({
+      productId: product.id,
+      slug: product.slug,
+      name: product.name,
+      vector: embedding,
+      updatedAt: new Date().toISOString(),
+    });
+
+    console.log(`Indexed successfully: ${product.name}`);
+    return { success: true, skipped: false };
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : String(error);
+    console.error(`Error processing ${product.id}:`, errorMessage);
+    return { success: false, skipped: false, error: errorMessage };
+  }
+}
+
+/**
+ * Process products in parallel with concurrency control
+ */
+async function processProductsInParallel(
```

Comment:
this won't work in parallel since it's not using Web Workers or node's worker-threads, only concurrently


it's fine since we can drop-in replace db for a real vector database, let's just change the name


```suggestion
/**
 * Process products in parallel with concurrency control
 */
async function processProductsConcurently(
```